posix/posix.c

```diff
@@ -266,21 +284,22 @@ static int pinfo_cmp(rbnode_t *n1, rbnode_t *n2)
 	else if (p1->process > p2->process) {
 		return 1;
 	}
-
-	return 0;
+	else {
+		return 0;
+	}
 }
 
 
-int posix_truncate(oid_t *oid, off_t length)
+static int posix_truncate(oid_t *oid, off_t length)
 {
 	msg_t msg;
 	int err = -EINVAL;
 
-	if ((oid->port != US_PORT) && (length >= 0)) {
+	if ((oid->port != US_PORT) && ((unsigned long long)length >= 0ULL)) {
```

Comment:
`((unsigned long long)length >= 0ULL)` makes no sense.
```suggestion
	if ((oid->port != US_PORT) && (length >= 0LL)) {
```